core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -202,8 +202,9 @@ private RexNode simplifyFilterCondition(RexNode condition, RelOptRuleCall call)
     // 1>null evaluates to unknown and WHERE unknown behaves exactly like WHERE false
     RexSimplify simplifier =
         new RexSimplify(xBuilder, RelOptPredicateList.EMPTY, executor);
-    return RexUtil.removeNullabilityCast(
+    RexNode simplified = RexUtil.removeNullabilityCast(
         xBuilder.getTypeFactory(), simplifier.simplifyUnknownAsFalse(condition));
+    return simplified.isAlwaysFalse() ? condition : simplified;
```

Comment:
Hi @danny0405 , without the check, tree may contain Filter with false condition, so then implementors (like Elasticsearch or MongoDB adapters) can't provide implementation for such Filter. The hidden issue was always there, just became visible now, after such simplification actually happened in tests.

Reply:
This change doesn't seem right. 
Maybe you should try to use FilterProjectTransposeRule with `copyFilter=false` ? Or maybe we should review this piece of code and, in case of an "always false condition" create an empty Values instead of a Filter (as RelBuilder#filter does).

Reply:
BTW, there was some discussion about this rule and the condition simplification in https://issues.apache.org/jira/browse/CALCITE-2582